src/cpu/instructions.rs

```diff
@@ -0,0 +1,641 @@
+            0xB8 => self.alu_a_r8(AluBinary::Cp, R8::B), // CP A, B
+            0xB9 => self.alu_a_r8(AluBinary::Cp, R8::C), // CP A, C
+            0xBA => self.alu_a_r8(AluBinary::Cp, R8::D), // CP A, D
+            0xBB => self.alu_a_r8(AluBinary::Cp, R8::E), // CP A, E
+            0xBC => self.alu_a_r8(AluBinary::Cp, R8::H), // CP A, H
+            0xBD => self.alu_a_r8(AluBinary::Cp, R8::L), // CP A, L
+            0xBE => self.alu_a_at_hl(AluBinary::Cp),     // CP A, [HL]
+            0xBF => self.alu_a_r8(AluBinary::Cp, R8::A), // CP A, A
+
+            0xC0 => self.ret_cc(Flag::Z, false),      // RET NZ
+            0xC1 => self.pop_r16(R16::BC),            // POP BC
+            0xC2 => self.jp_cc_a16(Flag::Z, false),   // JP NZ
+            0xC3 => self.jp_a16(),                    // JP a16
+            0xC4 => self.call_cc_a16(Flag::Z, false), // CALL NZ, a16
+            0xC5 => self.push_r16(R16::BC),           // PUSH BC
+            0xC6 => self.alu_a_n8(AluBinary::Add),    // ADD A, n8
+            0xC7 => self.rst_vec(0x00),               // RST $00
+            0xC8 => self.ret_cc(Flag::Z, true),       // RET Z
+            0xC9 => self.ret(),                       // RET
+            0xCA => self.jp_cc_a16(Flag::Z, true),    // JP Z, a16
+            0xCB => self.current_instruction_is_prefixed = true, // PREFIX
+            0xCC => self.call_cc_a16(Flag::Z, true),  // CALL Z, a16
+            0xCD => self.call_a16(),                  // CALL a16
+            0xCE => self.alu_a_n8(AluBinary::Adc),    // ADC A, n8
+            0xCF => self.rst_vec(0x08),               // RST $08
+
+            0xD0 => self.ret_cc(Flag::C, false),      // RET NC
+            0xD1 => self.pop_r16(R16::DE),            // POP DE
+            0xD2 => self.jp_cc_a16(Flag::C, false),   // JP NC, a16
+            0xD3 => panic!("Invalid Instruction"),    // ---
+            0xD4 => self.call_cc_a16(Flag::C, false), // CALL NC, a16
+            0xD5 => self.push_r16(R16::DE),           // PUSH DE
+            0xD6 => self.alu_a_n8(AluBinary::Sub),    // SUB A, n8
+            0xD7 => self.rst_vec(0x10),               // RST $10
+            0xD8 => self.ret_cc(Flag::C, true),       // RET C
+            0xD9 => self.reti(),                      // RETI
+            0xDA => self.jp_cc_a16(Flag::C, true),    // JP C, a16
+            0xDB => panic!("Invalid Instruction"),    // ---
+            0xDC => self.call_cc_a16(Flag::C, true),  // CALL C, a16
+            0xDD => panic!("Invalid Instruction"),    // ---
+            0xDE => self.alu_a_n8(AluBinary::Sbc),    // SBC A, n8
+            0xDF => self.rst_vec(0x18),               // RST $18
+
+            0xE0 => self.ldh_at_a8_a(),            // LDH [a8], A
+            0xE1 => self.pop_r16(R16::HL),         // POP HL
+            0xE2 => self.ldh_c_a(),                // LDH [C], A
+            0xE3 => panic!("Invalid Instruction"), // ---
```

Comment:
I was looking at this, thinking that these panic!() instructions are a great opportunity to return an Error type. Also, on this type of problem you might want to dump out the contents of the CPU state (registers, flags, PC, etc) to the console.